api/envoy/admin/v2alpha/clusters.proto

```diff
@@ -0,0 +1,93 @@
+syntax = "proto3";
+
+package envoy.admin.v2alpha;
+
+import "envoy/api/v2/core/base.proto";
+import "envoy/api/v2/core/address.proto";
+import "envoy/type/percent.proto";
+
+// [#protodoc-title: Clusters]
+
+// Admin endpoint uses this wrapper for `/clusters` to display cluster status information.
+// See :ref:`/clusters <operations_admin_interface_clusters>` for more information.
+message Clusters {
+  // Mapping from cluster name to each cluster's status.
+  repeated ClusterStatus cluster_statuses = 1;
+}
+
+// Details an individual cluster's current status.
+message ClusterStatus {
+  // Name of the cluster.
+  string name = 1;
+
+  // General outlier statistics if installed for this cluster.
+  OutlierInfo outlier_info = 2;
+
+  // Denotes whether this cluster was added via API or configured statically.
+  bool added_via_api = 3;
+
+  // Mapping from host address to the host's current status.
+  repeated HostStatus host_statuses = 4;
+}
+
+// :ref:`Cluster outlier detection <arch_overview_outlier_detection>` statistics. -1 for any of
+// these statistics denotes that there was not enough data to compute it.
+message OutlierInfo {
+  // The average success rate of the hosts in the Detector for the last aggregation interval.
+  //
+  // Note: the message will be omitted if there were not enough hosts with enough request volume to
+  // proceed with success rate based outlier ejection.
+  envoy.type.Percent success_rate_average = 1;
+
+  // The success rate threshold used in the last interval. The threshold is used to eject hosts
+  // based on their success rate.
+  //
+  // Note: the message will be omitted if there were not enough hosts with enough request volume to
+  // proceed with success rate based outlier ejection. Since this value can technically be negative
+  // due to the way that it's computed, negative values will also be represented by the absence of
+  // this field because a negative value implies that there was no threshold for that interval.
+  envoy.type.Percent success_rate_ejection_threshold = 2;
+}
+
+// Current state of a particular host.
+message HostStatus {
+  // Address of this host.
+  envoy.api.v2.core.Address address = 1;
+
+  // Mapping from the name of the statistic to the current value.
+  map<string, int64> stats = 2;
```

Comment:
Is this the first time we're defining in proto the stats data model?

Reply:
Not exactly - see [here](https://github.com/envoyproxy/envoy/blob/master/api/envoy/service/metrics/v2/metrics_service.proto#L36).  The only more sophisticated definition I could imagine would be differentiating between gauges and counters.

Reply:
OK, I think it's fine to have a simpler variant without relying on Prometheus protos. Might belong in its own message though, so we can reuse again later. Would suggest the domain to be `uint64`.

Reply:
SGTM.  Where do you think this stat message should go?  under `api/envoy/data/`, in `api/envoy/metrics/v2/stats.proto`, or somewhere else?